modules/evm/src/lib.rs

```diff
@@ -571,7 +571,32 @@ pub mod module {
 				gas_limit,
 				storage_limit,
 				T::config(),
-			)?;
+			)
+			.map_err(|e| {
+				Pallet::<T>::deposit_event(Event::<T>::ExecutedFailed {
+					from: source,
+					contract: target,
+					exit_reason: ExitReason::Error(ExitError::Other(Into::<&str>::into(e).into())),
+					output: vec![],
+					logs: vec![],
+				});
+				e
+			})?;
```

Comment:
This is for deposit event then rollback transaction.

Reply:
events are stored in storage so rollback will also remove the events.
so those extrinsic should not fail, they should always have ExtrinsicSuccess event but with another event to indicate the result.